plugins/junit/src/com/intellij/execution/junit/codeInsight/JUnit5MalformedOrderedTestInspection.kt

```diff
@@ -0,0 +1,102 @@
+import com.intellij.codeInspection.ProblemDescriptor
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.openapi.project.Project
+import com.intellij.psi.*
+import com.intellij.psi.codeStyle.CodeStyleManager
+import com.intellij.psi.codeStyle.JavaCodeStyleManager
+import com.intellij.psi.impl.PsiClassImplUtil
+import com.intellij.psi.util.PsiTreeUtil
+import com.intellij.psi.util.PsiTypesUtil
+import com.intellij.psi.util.PsiUtil
+import com.siyeh.ig.junit.JUnitCommonClassNames
+
+class JUnit5MalformedOrderedTestInspection : AbstractBaseJavaLocalInspectionTool() {
+  override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor {
+    val file = holder.file
+    if (JavaPsiFacade.getInstance(file.project).findClass(JUnitCommonClassNames.ORG_JUNIT_JUPITER_API_TEST, file.resolveScope) == null) {
+      return PsiElementVisitor.EMPTY_VISITOR
+    }
+    val psiClassOrderAnnotationOrderer = JavaPsiFacade.getInstance(file.project)
+                                           .findClass("org.junit.jupiter.api.MethodOrderer.OrderAnnotation", file.resolveScope)
+                                         ?: return PsiElementVisitor.EMPTY_VISITOR
+    return object : JavaElementVisitor() {
+      override fun visitMethod(method: PsiMethod) {
+
+        AnnotationUtil.findAnnotation(method, JUnitCommonClassNames.ORG_JUNIT_JUPITER_API_TEST) ?: return
+        val orderAnnotation = AnnotationUtil.findAnnotation(method, "org.junit.jupiter.api.Order") ?: return
+        val psiClass = method.containingClass ?: return
+
+        val testMethodOrderAnnotation = AnnotationUtil.findAnnotation(psiClass, "org.junit.jupiter.api.TestMethodOrder")
+        if (testMethodOrderAnnotation == null) {
+          holder.registerProblem(orderAnnotation, "Method with @Order inside class without @TestMethodOrder", AddAnnotationFix())
+          return
+        }
+
+        val value = testMethodOrderAnnotation.findAttributeValue("value")
+                      as PsiClassObjectAccessExpression? ?: return
+
+        val valuePsiClass = PsiTypesUtil.getPsiClass(value.operand.type) ?: return
+        if (PsiClassImplUtil.isClassEquivalentTo(valuePsiClass, psiClassOrderAnnotationOrderer)) {
+          // annotation value is known correct MethodOrderer implementation
+          return
+        }
+
+        val packageName = PsiUtil.getPackageName(valuePsiClass) ?: return
+        if (packageName.startsWith("org.junit.jupiter.api")) {
+          // other JUnit5 MethodOrderer implementations ignore @Order annotation
+          holder.registerProblem(orderAnnotation,
+                                 "Method with @Order inside class with @TestMethodOrder but without @Order support",
+                                 ReplaceAnnotationFix())
+        }
+        // but custom implementation of MethodOrderer may take @Order annotations into account
+      }
+    }
+  }
+}
+
+private abstract class AnnotationFix : LocalQuickFix {
```

Comment:
`com.intellij.codeInsight.intention.AddAnnotationFix`?

Reply:
i am goning to investigate usage and try to replace it with my `AddAnnotationFix`/`ReplaceAnnotationFix`

Reply:
I have replaced my implementations with one you mentioned.